lib/src/path_parser.cc

```diff
@@ -143,14 +143,18 @@ namespace hocon {
             return token_list { t };
         }
 
+        // The split iterator tacks an empty string onto the end of the iterator if the string
+        // ends in the split character; so we need to remove trailing '.'
+        string text_to_split = boost::trim_right_copy_if(token_text, is_any_of("."));
```

Comment:
Ah, I misunderstood what it was checking (`!=`, not `==`). Also double checked Java's [string.split](http://docs.oracle.com/javase/7/docs/api/java/lang/String.html#split(java.lang.String)) implementation to verify it leaves off trailing empty strings.